### ISSUE TITLE
The error page handler didn't process correctly in proxy

### ISSUE THREAD

**Issue by gaoxingliang** [eclipse/jetty.project#3806]
Hi, I’m using Jetty 9.4.18.v20190429 to implement the reverse proxy function.
but it has a problem in Chrome 73 when customizing the error page.

Case:
Chrome added a new header application/signed-exchange;v=b3 and this is not a text/ style header since version 73. and this cause a different behavior with the firefox.
On chrome it’s (note the content-length is 0)
![image](https://user-images.githubusercontent.com/3148215/59984931-af161e80-9660-11e9-84e5-1dc893c4382b.png)
![image](https://user-images.githubusercontent.com/3148215/59984939-bc330d80-9660-11e9-9789-6c2e6c789e16.png)
On firefox:
![image](https://user-images.githubusercontent.com/3148215/59984952-dec52680-9660-11e9-9615-8097d1431c3c.png)

after digging more:
Code analysis:

`org.eclipse.jetty.server.handler.ErrorHandler#generateAcceptableResponse(org.eclipse.jetty.server.Request, javax.servlet.http.HttpServletRequest, javax.servlet.http.HttpServletResponse, int, java.lang.String)`
```
        List<String> acceptable=baseRequest.getHttpFields().getQualityCSV(HttpHeader.ACCEPT, QuotedQualityCSV.MOST_SPECIFIC_MIME_ORDERING);
        
        if (acceptable.isEmpty() && !baseRequest.getHttpFields().contains(HttpHeader.ACCEPT))
        {
            generateAcceptableResponse(baseRequest,request,response,code,message,MimeTypes.Type.TEXT_HTML.asString());
        }
        else
        {
            for (String mimeType:acceptable)
            {
                generateAcceptableResponse(baseRequest,request,response,code,message,mimeType);
                if (baseRequest.isHandled())
                    break;
            }
        }
        baseRequest.setHandled(true);
        baseRequest.getResponse().closeOutput();
```
this method has a loop on the accept headers to process and generate a readable message.
but this will NOT work because the 
`baseRequest.isHandled()  is always true when entering this code`.

**Comment by janbartel** [comment]
@gaoxingliang can you provide a small test harness that demonstrates the problem, and also full http dialogs?

**Comment by janbartel** [comment]
@gaoxingliang I've tried this header in the jetty test webapp and jetty is handling this correctly:  the `ErrorHandler.generateAcceptableResponse(Request baseRequest, HttpServletRequest request, HttpServletResponse response, int code, String message)` method is called, which extracts all of the Accept headers in order of quality, which is (as per the spec https://tools.ietf.org/html/rfc7231#section-5.3.2):

```
application/signed-exchange;v=b3
text/html
application/xhtml+xml
image/webp
image/apng
application/xml
*/*
```

Thus, jetty then calls `ErrorHandler.generateAcceptableResponse(Request baseRequest, HttpServletRequest request, HttpServletResponse response, int code, String message, String mimeType)` with `application/signed-exchange;v=b3`. As you can see, this is not one of the mimetypes handled by that method, so the loops continues around to `text/html`, which is handled.

I suspect you have overridden the `ErrorHandler.generateAcceptableResponse(Request baseRequest, HttpServletRequest request, HttpServletResponse response, int code, String message, String mimeType)` method incorrectly.

**Comment by gaoxingliang** [comment]
@janbartel 
Here is the full demo code: https://github.com/gaoxingliang/goodutils/blob/master/proxydemeo/JettyIssue3806.MD

**Comment by gregw** [comment]
@janbartel has been able to reproduce the problem.  This appears to be a race condition with the isHandled flag which is set to true then false by the async dispatch cycle of the ProxyServlet.   So sendError is being called from a callback and it depends on if this happens before/after the original dispatch returns.   Thus because it is valid to call `sendError` asynchronously, it is probably not suitable to use isHandled as the flag within the errorHandler to determine if handling has been achieved - However, I do have concerns that sendError might actually perform a dispatch while another dispatch is underway.

@gaoxingliang thanks for persisting with this, you have definitely revealed a bug!

**Comment by gregw** [comment]
Fully async fix in 9.4.21
